docs/source/elasticsearch.asciidoc

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2024 Timeseer.AI
+// SPDX-License-Identifier: Apache-2.0
+
+== Elasticsearch
+
+Sources with `type = "elasticsearch"` configure https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html[Elasticsearch] sources.
+
+
+```toml
+[source.<name>]
+type = "elasticsearch"
+scheme = "http://"
+host = "localhost"
+port = 9200
+index = ""
+list_index = ""
```

Comment:
To be consistent with the CSV source, we should rename this to `metadata_index`.